tests/generic_config_updater/test_dhcp_relay.py

```diff
@@ -0,0 +1,450 @@
+    """
+    if init_dhcp_server_config != "default_setup":
+        pytest.skip("Unsupported init config")
+
+    dhcp_rm_nonexist_json = [
+        {
+            "op": "remove",
+            "path": "/VLAN/Vlan"+ str(vlan_intfs_list[0]) + "/dhcp_servers/5"
+        }]
+
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {}".format(tmpfile))
+
+    output = apply_patch(duthost, json_data=dhcp_rm_nonexist_json, dest_file=tmpfile)
+    expect_op_failure(output)
+
+    delete_tmpfile(duthost, tmpfile)
+
+def test_dhcp_relay_tc4_add_exist(duthost, setup_vlan, init_dhcp_server_config, vlan_intfs_list):
+    """Test add existed dhcp server on default setup
+    """
+    if init_dhcp_server_config != "default_setup":
+        pytest.skip("Unsupported init config")
+
+    dhcp_add_exist_json = [
+        {
+            "op": "add",
+            "path": "/VLAN/Vlan" + str(vlan_intfs_list[0]) + "/dhcp_servers/2",
+            "value": "192.0." + str(vlan_intfs_list[0]) + ".3"
+        }]
+
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {}".format(tmpfile))
+
+    output = apply_patch(duthost, json_data=dhcp_add_exist_json, dest_file=tmpfile)
+    expect_op_failure(output)
+
+    delete_tmpfile(duthost, tmpfile)
+
+def test_dhcp_relay_tc5_rm(duthost, setup_vlan, init_dhcp_server_config, vlan_intfs_list):
+    """Test normal remove dhcp server on default setup
+
+    This VLAN detail should show below after test
+    +-----------+------------------+-----------+----------------+-------------+-----------------------+
+    |   VLAN ID | IP Address       | Ports     | Port Tagging   | Proxy ARP   | DHCP Helper Address   |
+    +===========+==================+===========+================+=============+=======================+
+    |       108 | 192.168.108.1/24 | Ethernet4 | tagged         | disabled    | 192.0.108.1           |
+    |           |                  |           |                |             | 192.0.108.2           |
+    |           |                  |           |                |             | 192.0.108.3           |
+    +-----------+------------------+-----------+----------------+-------------+-----------------------+
+    |       109 | 192.168.109.1/24 | Ethernet4 | tagged         | disabled    | 192.0.109.1           |
+    |           |                  |           |                |             | 192.0.109.2           |
+    |           |                  |           |                |             | 192.0.109.3           |
+    |           |                  |           |                |             | 192.0.109.4           |
+    +-----------+------------------+-----------+----------------+-------------+-----------------------+
+    """
+    if init_dhcp_server_config != "default_setup":
+        pytest.skip("Unsupported init config")
+
+    dhcp_rm_json = [
+        {
+        "op": "remove",
+        "path": "/VLAN/Vlan" + str(vlan_intfs_list[0]) + "/dhcp_servers/3"
```

Comment:
>3 [](http://example.com/codeflow?start=72&length=1)

also a magic number

Reply:
Here I use magic number due that if the index change, the json will look much different. Here I choose to remove the last element for simplicity but also check the correctness. If I choose to remove other index such as 1. The json_patch will look like this:
```
[
 {
  "op": "remove",
  "path": "/VLAN/Vlan108/dhcp_servers/3"
 },
 {
  "op": "replace",
  "path": "/VLAN/Vlan108/dhcp_servers/2",
  "value": "192.0.108.4"
 },
 {
  "op": "replace",
  "path": "/VLAN/Vlan108/dhcp_servers/1",
  "value": "192.0.108.3"
 }
]
```
The vlan dhcp server is not simple `key->value` pair. It looks like a `key->indexed value`